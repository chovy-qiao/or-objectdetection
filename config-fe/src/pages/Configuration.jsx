import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import {
  Grid,
  Container,
  Typography,
  Card,
  TextField,
  FormGroup,
  FormControlLabel,
  Checkbox,
  Button,
  Chip,
  CardContent,
  FormControl,
  MenuItem,
  InputLabel,
  Box
} from "@material-ui/core";
import ReactPlayer from "react-player";
import Canvas from "../features/custom/Canvas";
import Editor from "../features/custom/Editor";

const Configuration = (props) => {
  const classes = useStyles();
  const [state, setState] = React.useState({
    gilad: false,
    jason: false,
    antoine: false,
    bert: false,
    jan: false,
    peer: false,
    johan: false
  });

  const canvasprops = {
    color: "rgb(56, 255, 20, 80%)",
    width: "640px",
    height: "360px",
    brushRadius: 3,
    lazyRadius: 8
  };

  const handleChange = (event) => {
    setState({ ...state, [event.target.name]: event.target.checked });
  };

  const { D, BB, CLC, CO, SOS, VC } = state;

  const handleDelete = (chipToDelete) => () => {
    setChipData((chips) =>
      chips.filter((chips) => chips.key !== chipToDelete.key)
    );
  };

  const handleSelect = (e) => {
    setChipData([...chipData, { key: e.target.value, label: e.target.value }]);
  };

  const id = props.match.params.id;

  const [chipData, setChipData] = React.useState([]);

  const detectionOptions = [
    { key: 1, label: "Person" },
    { key: 2, label: "Car" }
  ];

  return (
    <div>
      <Grid container spacing={3} style={{ margin: "1%" }}>
        <Grid item xs={6}>
          <Typography variant="h3" className={classes.title}>
            Configuratie - Camera X
          </Typography>

          <Card className={classes.root}>
            <CardContent className={classes.cardContent}>
              <Box className={classes.box}>
                <div style={{ display: "inline-block" }}>
                  <InputLabel className={classes.inputTitle}>
                    <b>Naam:</b>
                  </InputLabel>
                  <Typography
                    className={classes.title2}
                    color="textSecondary"
                    gutterBottom
                  >
                    <TextField
                      size="normal"
                      id="standard-basic"
                      value="Kruispunt XY"
                    />
                  </Typography>
                </div>
                <div style={{ display: "inline-block" }}>
                  <InputLabel className={classes.inputTitle}>
                    <b>Locatie:</b>
                  </InputLabel>
                  <Typography
                    className={classes.title2}
                    color="textSecondary"
                    gutterBottom
                  >
                    <TextField
                      size="normal"
                      id="standard-basic"
                      value="Strijp S"
                    />
                  </Typography>
                </div>
                <div style={{ display: "inline-block" }}>
                  <InputLabel className={classes.inputTitle}>
                    <b>Type:</b>
                  </InputLabel>
                  <Typography
                    className={classes.title2}
                    color="textSecondary"
                    gutterBottom
                  >
                    <TextField
                      size="normal"
                      id="standard-basic"
                      value="Camera C920"
                      disabled
                    />
                  </Typography>
                </div>
                <div style={{ display: "inline-block" }}>
                  <InputLabel className={classes.inputTitle}>
                    <b>Resolutie:</b>
                  </InputLabel>
                  <Typography
                    className={classes.title2}
                    color="textSecondary"
                    gutterBottom
                  >
                    <TextField
                      size="normal"
                      id="standard-basic"
                      value="1920x1080"
                      disabled
                    />
                  </Typography>
                </div>
                <div style={{ display: "inline-block" }}>
                  <InputLabel className={classes.inputTitle}>
                    <b>Framerate:</b>
                  </InputLabel>
                  <Typography
                    className={classes.title2}
                    color="textSecondary"
                    gutterBottom
                  >
                    <TextField
                      size="normal"
                      id="standard-basic"
                      value="30fps"
                      disabled
                    />
                  </Typography>
                </div>
                <InputLabel className={classes.inputTitle}>
                  <b>Url:</b>
                </InputLabel>
                <Typography
                  className={classes.title}
                  color="textSecondary"
                  gutterBottom
                >
                  <TextField
                    className={classes.textField}
                    size="normal"
                    id="standard-basic"
                    value="https://www.openremote.com/videofeed/kruispunt"
                    disabled
                  />
                </Typography>
                <InputLabel className={classes.inputTitle}>
                  <b>Selecteer Detecties</b>
                </InputLabel>
                <TextField
                  className={classes.comboField}
                  select
                  size="normal"
                  value=""
                  onChange={(e) => handleSelect(e)}
                >
                  {detectionOptions.map((option) => (
                    <MenuItem key={option.key} value={option.label}>
                      {option.label}
                    </MenuItem>
                  ))}
                </TextField>
                <ul className={classes.chipControl}>
                  {chipData.map((data, index) => {
                    return (
                      <Chip
                        className={classes.chip}
                        key={data.key + index}
                        label={data.label}
                        onDelete={handleDelete(data)}
                        color="primary"
                      />
                    );
                  })}
                </ul>
                <InputLabel className={classes.inputTitle}>
                  <b>Detectie Opties</b>
                </InputLabel>
                <FormControl
                  component="fieldset"
                  className={classes.formControl}
                >
                  <FormGroup>
                    <FormControlLabel
                      control={
                        <Checkbox
                          checked={D}
                          onChange={handleChange}
                          name="D"
                          style={{
                            color: "#4D9D2A"
                          }}
                        />
                      }
                      label="Directions"
                    />
                    <FormControlLabel
                      control={
                        <Checkbox
                          checked={BB}
                          onChange={handleChange}
                          name="BB"
                          style={{
                            color: "#4D9D2A"
                          }}
                        />
                      }
                      label="Boundary Boxes"
                    />
                    <FormControlLabel
                      control={
                        <Checkbox
                          checked={CLC}
                          onChange={handleChange}
                          name="CLC"
                          style={{
                            color: "#4D9D2A"
                          }}
                        />
                      }
                      label="Calculate Line Cross"
                    />
                    <FormControlLabel
                      control={
                        <Checkbox
                          checked={CO}
                          onChange={handleChange}
                          name="CO"
                          style={{
                            color: "#4D9D2A"
                          }}
                        />
                      }
                      label="Count Objects"
                    />
                    <FormControlLabel
                      control={
                        <Checkbox
                          checked={SOS}
                          onChange={handleChange}
                          name="SOS"
                          style={{
                            color: "#4D9D2A"
                          }}
                        />
                      }
                      label="Show object speed"
                    />
                    <FormControlLabel
                      control={
                        <Checkbox
                          checked={VC}
                          onChange={handleChange}
                          name="VC"
                          style={{
                            color: "#4D9D2A"
                          }}
                        />
                      }
                      label="Visualize Centers"
                    />
                    <Button
                      variant="contained"
                      color="primary"
                      className={classes.Buttons}
                    >
                      Save
                    </Button>
                    <Button
                      variant="contained"
                      color="default"
                      className={classes.Buttons}
                    >
                      Cancel
                    </Button>
                  </FormGroup>
                </FormControl>
              </Box>
            </CardContent>
          </Card>
        </Grid>
        <Grid item xs={6}>
          <Typography variant="h3" className={classes.title}>
            Camera Preview
          </Typography>
          <img src="https://static.dw.com/image/47113704_303.jpg" />
          <Editor />
        </Grid>
      </Grid>
    </div>
  );
};

export default Configuration;

const useStyles = makeStyles((theme) => ({
  formControl: {
    margin: theme.spacing(1),
    width: "100%",
    display: "flex"
  },
  selectEmpty: {
    marginTop: theme.spacing(2)
  },
  title: {
    color: "#4D9D2A",
    marginBottom: "20px"
  },
  title2: {
    color: "#4D9D2A",
    marginBottom: "20px",
    width: "100%",
    display: "inline-block",
    marginRight: 40
  },
  inputTitle: {
    marginTop: "30px",
    width: "40%"
  },
  Buttons: {
    marginTop: "10px",
    width: "100%"
  },
  chipControl: {
    display: "flex",
    justifyContent: "left",
    flexWrap: "wrap",
    listStyle: "none",
    marginTop: 10,
    maxWidth: 300,
    padding: 0
  },
  chip: {
    margin: theme.spacing(0.5)
  },
  inputField: {
    height: 80
  },
  box: {
    margin: "10px",
    width: "100%"
  },
  root: {
    width: "100%"
  },
  textField: {
    width: "100%"
  },
  comboField: {
    width: "50%"
  },
  cardContent: {
    display: "flex"
  }
}));
