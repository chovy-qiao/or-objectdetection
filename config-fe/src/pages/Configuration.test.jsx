import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Configuration from "./Configuration";
import { MemoryRouter } from "react-router-dom";

const renderConfiguration = ({ match = { params: { id: "aap" } } } = {}) => {
  return render(
    <MemoryRouter>
      <Configuration match={match} />
    </MemoryRouter>
  );
};
describe("Configuration component", () => {
  it("matches the snapshot", () => {
    const { container } = renderConfiguration();
    expect(container).toMatchSnapshot();
  });
});
