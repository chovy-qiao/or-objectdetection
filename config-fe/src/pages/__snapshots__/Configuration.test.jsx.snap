// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration component matches the snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      style="margin: 1%;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <h3
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h3"
        >
          Configuratie - Camera X
        </h3>
        <div
          class="MuiPaper-root MuiCard-root makeStyles-root-11 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root makeStyles-cardContent-14"
          >
            <div
              class="MuiBox-root MuiBox-root-15 makeStyles-box-10"
            >
              <div
                style="display: inline-block;"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputTitle-5 MuiInputLabel-animated"
                >
                  <b>
                    Naam:
                  </b>
                </label>
                <p
                  class="MuiTypography-root makeStyles-title2-4 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        id="standard-basic"
                        type="text"
                        value="Kruispunt XY"
                      />
                    </div>
                  </div>
                </p>
              </div>
              <div
                style="display: inline-block;"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputTitle-5 MuiInputLabel-animated"
                >
                  <b>
                    Locatie:
                  </b>
                </label>
                <p
                  class="MuiTypography-root makeStyles-title2-4 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        id="standard-basic"
                        type="text"
                        value="Strijp S"
                      />
                    </div>
                  </div>
                </p>
              </div>
              <div
                style="display: inline-block;"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputTitle-5 MuiInputLabel-animated"
                >
                  <b>
                    Type:
                  </b>
                </label>
                <p
                  class="MuiTypography-root makeStyles-title2-4 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
                        disabled=""
                        id="standard-basic"
                        type="text"
                        value="Camera C920"
                      />
                    </div>
                  </div>
                </p>
              </div>
              <div
                style="display: inline-block;"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputTitle-5 MuiInputLabel-animated"
                >
                  <b>
                    Resolutie:
                  </b>
                </label>
                <p
                  class="MuiTypography-root makeStyles-title2-4 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
                        disabled=""
                        id="standard-basic"
                        type="text"
                        value="1920x1080"
                      />
                    </div>
                  </div>
                </p>
              </div>
              <div
                style="display: inline-block;"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputTitle-5 MuiInputLabel-animated"
                >
                  <b>
                    Framerate:
                  </b>
                </label>
                <p
                  class="MuiTypography-root makeStyles-title2-4 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
                        disabled=""
                        id="standard-basic"
                        type="text"
                        value="30fps"
                      />
                    </div>
                  </div>
                </p>
              </div>
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputTitle-5 MuiInputLabel-animated"
              >
                <b>
                  Url:
                </b>
              </label>
              <p
                class="MuiTypography-root makeStyles-title-3 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root makeStyles-textField-12"
                >
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
                      disabled=""
                      id="standard-basic"
                      type="text"
                      value="https://www.openremote.com/videofeed/kruispunt"
                    />
                  </div>
                </div>
              </p>
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputTitle-5 MuiInputLabel-animated"
              >
                <b>
                  Selecteer Detecties
                </b>
              </label>
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-comboField-13"
              >
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                >
                  <div
                    aria-haspopup="listbox"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    role="button"
                    tabindex="0"
                  >
                    <span>
                      â€‹
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
              <ul
                class="makeStyles-chipControl-7"
              />
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputTitle-5 MuiInputLabel-animated"
              >
                <b>
                  Detectie Opties
                </b>
              </label>
              <fieldset
                class="MuiFormControl-root makeStyles-formControl-1"
              >
                <div
                  class="MuiFormGroup-root"
                >
                  <label
                    class="MuiFormControlLabel-root"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                      style="color: rgb(77, 157, 42);"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-19"
                          data-indeterminate="false"
                          name="D"
                          type="checkbox"
                          value=""
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Directions
                    </span>
                  </label>
                  <label
                    class="MuiFormControlLabel-root"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                      style="color: rgb(77, 157, 42);"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-19"
                          data-indeterminate="false"
                          name="BB"
                          type="checkbox"
                          value=""
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Boundary Boxes
                    </span>
                  </label>
                  <label
                    class="MuiFormControlLabel-root"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                      style="color: rgb(77, 157, 42);"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-19"
                          data-indeterminate="false"
                          name="CLC"
                          type="checkbox"
                          value=""
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Calculate Line Cross
                    </span>
                  </label>
                  <label
                    class="MuiFormControlLabel-root"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                      style="color: rgb(77, 157, 42);"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-19"
                          data-indeterminate="false"
                          name="CO"
                          type="checkbox"
                          value=""
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Count Objects
                    </span>
                  </label>
                  <label
                    class="MuiFormControlLabel-root"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                      style="color: rgb(77, 157, 42);"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-19"
                          data-indeterminate="false"
                          name="SOS"
                          type="checkbox"
                          value=""
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Show object speed
                    </span>
                  </label>
                  <label
                    class="MuiFormControlLabel-root"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                      style="color: rgb(77, 157, 42);"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-19"
                          data-indeterminate="false"
                          name="VC"
                          type="checkbox"
                          value=""
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Visualize Centers
                    </span>
                  </label>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-Buttons-6 MuiButton-containedPrimary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Save
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-Buttons-6"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Cancel
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <h3
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h3"
        >
          Camera Preview
        </h3>
        <img
          src="https://static.dw.com/image/47113704_303.jpg"
        />
        <div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Open Editor
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
